src/state/hooks.ts

```diff
@@ -99,6 +107,75 @@ export const usePoolFromPid = (sousId: number): Pool => {
   return transformPool(pool)
 }
 
+export const useFetchCakeVault = () => {
+  const { account } = useWeb3React()
+  const { fastRefresh } = useRefresh()
+  const dispatch = useAppDispatch()
+
+  useEffect(() => {
+    dispatch(fetchCakeVaultPublicData())
+  }, [dispatch, fastRefresh])
+
+  useEffect(() => {
+    dispatch(fetchCakeVaultUserData({ account }))
+  }, [dispatch, fastRefresh, account])
+
+  useEffect(() => {
+    dispatch(fetchCakeVaultFees())
+  }, [dispatch])
+}
+
+export const useCakeVault = () => {
+  const {
+    totalShares: totalSharesAsString,
+    pricePerFullShare: pricePerFullShareAsString,
+    totalCakeInVault: totalCakeInVaultAsString,
+    estimatedCakeBountyReward: estimatedCakeBountyRewardAsString,
+    totalPendingCakeHarvest: totalPendingCakeHarvestAsString,
+    fees: { performanceFee, callFee, withdrawalFee, withdrawalFeePeriod },
+    userData: {
+      isLoading,
+      userShares: userSharesAsString,
+      cakeAtLastUserAction: cakeAtLastUserActionAsString,
+      lastDepositedTime,
+      lastUserActionTime,
+    },
+  } = useSelector((state: State) => state.pools.cakeVault)
+
+  // TODO: Figure out why this needs to be memoised to prevent rerender loop within BountyCard effect
```

Comment:
I don't really know why, but when this value isn't memoised... this effect within `BountyCard` creates an infinite loop:

https://github.com/pancakeswap/pancake-frontend/pull/1208/files#diff-80c27d122268c36a58ec1320fde37aaf56dcf8085b3a08450a164d04dc2d1ef7R43-R45